src/fusion_power_plant.h

```diff
@@ -59,27 +56,170 @@ class FusionPowerPlant : public cyclus::Facility  {
+    "tooltip": "Defaults to 100g/purchase", \
+    "units": "kg", \
+    "uitype": "range", \
+    "range": [0, 1e299], \
+    "uilabel": "Buy quantity" \
+  }
+  double buy_quantity;
+
+  #pragma cyclus var { \
+    "doc": "Helium-3 output commodity Designation", \
+    "tooltip": "He-3 output commodity", \
+    "uilabel": "He-3 output commodity" \
+  }
+  std::string he3_outcommod;
+
+  #pragma cyclus var { \
+    "doc": "Recipe for blanket feed material", \
+    "tooltip": "Recipe for blanket feed material", \
+    "uilabel": "Recipe for blanket feed material" \
+  }
+  std::string blanket_inrecipe;
+
+  #pragma cyclus var { \
+    "doc": "Blanket waste commodity designation", \
+    "tooltip": "Blanket waste commodity", \
+    "uilabel": "Blanket waste commodity" \
+  }
+  std::string blanket_outcommod;
+
+  #pragma cyclus var { \
+    "default": 1000.0, \
+    "doc": "Initial mass of full blanket material", \
+    "tooltip": "Only blanket material mass, not structural mass", \
+    "units": "kg", \
+    "uitype": "range", \
+    "range": [0, 10000], \
+    "uilabel": "Initial Mass of Blanket" \
+  }
+  double blanket_size;
+
+  #pragma cyclus var { \
+    "default": 0.05, \
+    "doc": "Percent of blanket that gets recycled every blanket turnover period", \
+    "tooltip": "Defaults to 0.05 (5%), must be between 0 and 15%", \
+    "units": "dimensionless", \
+    "uitype": "range", \
+    "range": [0, 0.15], \
+    "uilabel": "Blanket Turnover Rate" \
+  }
+  double blanket_turnover_rate;
+
+  //Resource Buffers and Trackers:
   cyclus::toolkit::ResBuf<cyclus::Material> tritium_storage;
   cyclus::toolkit::ResBuf<cyclus::Material> tritium_excess;
   cyclus::toolkit::ResBuf<cyclus::Material> helium_excess;
   cyclus::toolkit::ResBuf<cyclus::Material> blanket_feed;
   cyclus::toolkit::ResBuf<cyclus::Material> blanket_waste;
 
-  //PyneIDs
-  const int tritium_id = 10030000;
+  cyclus::toolkit::MatlBuyPolicy fuel_startup_policy;
+  cyclus::toolkit::MatlBuyPolicy fuel_refill_policy;
+  cyclus::toolkit::MatlBuyPolicy blanket_fill_policy;
 
-  //Compositions:
-  const cyclus::CompMap T = {{tritium_id, 1}};
-  const cyclus::Composition::Ptr tritium_comp = cyclus::Composition::CreateFromAtom(T);
+  cyclus::toolkit::MatlSellPolicy tritium_sell_policy;
+  cyclus::toolkit::MatlSellPolicy helium_sell_policy;
+  cyclus::toolkit::MatlSellPolicy blanket_waste_sell_policy;
 
-  //Materials:
-  cyclus::Material::Ptr sequestered_tritium = cyclus::Material::CreateUntracked(0.0, tritium_comp);
+  cyclus::toolkit::TotalInvTracker fuel_tracker;
+  cyclus::toolkit::TotalInvTracker blanket_tracker;
+
+  //Functions:
+  void CycleBlanket();
+  bool CheckOpeartingConditions();
+  bool CheckInitialStartupCondition();
+  void SequesterTritium();
+  void OperateReactor();
+  void CycleBlanket();
+  void DecayInventories();
+  void ExtractHelium();
+  void MoveExcessTritiumToSellBuffer();
+
+  private:
+    bool initial_startup_condition_met = false;
```

Comment:
Since you have a function to do this, I don't think you will need to save this state.  Rather, that function will test a variety of conditions each time and return true/false.

Reply:
So this one's actually a little tricky. Because the initial startup inventory condition is unique and only needs to be met once at the beginning of operation, I believe we do in fact have to store that it was met. If we continue to test for it after that initial startup it should (rightly) fail. This has to do with the startup inventory being greater than the operational inventory (the initial sequestration takes a big hit on the on hand tritium on the first timestep). We want to make sure that on that first operational timestep we have `startup_inventory` amount of tritium, but after that we only need to make sure that we have `reserve_inventory` amount, or more importantly enough on hand to operate for a full timestep.